Dockerfile

```diff
@@ -1,57 +1,50 @@
-FROM node:18 AS builder
-
-RUN apt-get update && apt-get install -y \
-    php \
-    php-cli \
-    php-zip \
-    php-xml \
-    php-mbstring \
-    php-curl \
-    php-sqlite3 \
-    php-ldap \
-    unzip \
-    cron
-
-RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
-
+####################
+# Build image
+####################
+FROM node:23-alpine AS builder
 WORKDIR /app
 
-COPY . .
+ADD js/ ./js/
+ADD css/ ./css/
+ADD yarn.lock .
+ADD package.json .
+ADD build.sh .
 
-RUN composer install
 RUN yarn install
 
 RUN chmod +x build.sh
 RUN ./build.sh
 
+####################
+# Runtime image
+####################
+FROM cirrusid/simplesamlphp:v2.3.7 AS runtime
 
-RUN chmod +x setup-php.sh
-RUN ./setup-php.sh
-
-FROM php:8.2-apache
-
-COPY --from=builder /app/ /var/www/html/
-
-RUN chown -R www-data:www-data /var/www/html \
-    && chmod -R 755 /var/www/html
+ENV SSP_NEW_UI=true
 
-RUN ln -s simplesamlphp/public simplesaml
-RUN mkdir -p /var/cache/simplesamlphp/core
-RUN chown -R www-data:www-data /var/cache/simplesamlphp
-RUN chmod -R 755 /var/cache/simplesamlphp
+# Install SimpleSAMLphp modules
+WORKDIR ${SSP_DIR}
+RUN composer config prefer-stable true \
+        && composer require --update-no-dev cirrusidentity/simplesamlphp-module-authoauth2
```

Comment:
I would recommend to put this in a `composer.json` and `composer.lock` file to prevent accidental breaking updates.

Reply:
Composer.lock would generate a ~4500 line file for a single package..
Instead, hard pinned the version in the Dockerfile now.